source/kraken/realtime.cpp

```diff
@@ -95,7 +114,61 @@ create_disruption(const std::string& id,
         impact->created_at = timestamp;
         impact->updated_at = timestamp;
         impact->application_periods.push_back(execution_period(circulation_date, mvj));
-        impact->severity = make_no_service_severity(timestamp, holder);
+        std::string wording;
+        nt::disruption::Effect effect;
+        if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_CANCELED) {
+            LOG4CPLUS_TRACE(log, "Disruption has NO_SERVICE effect");
+            // Yeah, that's quite hardcodded...
+            wording = "trip canceled!";
+            effect = nt::disruption::Effect::NO_SERVICE;
+        }
+        else if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_SCHEDULED
+                && trip_update.stop_time_update_size()) {
+            LOG4CPLUS_TRACE(log, "Disruption has MODIFIED_SERVICE effect");
+            // Yeah, that's quite hardcodded...
+            wording = "trip modified!";
+            effect = nt::disruption::Effect::MODIFIED_SERVICE;
+            LOG4CPLUS_TRACE(log, "Adding stop time into impact");
+            for (const auto& st: trip_update.stop_time_update()) {
+                auto* stop_point_ptr = data.pt_data->stop_points_map[st.stop_id()];
+                assert(stop_point_ptr);
+                uint32_t arrival_time = st.arrival().time();
+                uint32_t departure_time = st.departure().time();
+                LOG4CPLUS_TRACE(log, "arrival_time: " + std::to_string(arrival_time));
+                LOG4CPLUS_TRACE(log, "has_arrival_time: " + std::to_string(st.arrival().has_time()));
+
+                LOG4CPLUS_TRACE(log, "departure_time: " + std::to_string(departure_time));
+                LOG4CPLUS_TRACE(log, "has_departure_time: " + std::to_string(st.departure().has_time()));
+                {
+                    /*
+                     * When st.arrival().has_time() is false, st.arrival().time() will return 0, which, for kraken, is
+                     * considered as 00:00(midnight), This is not good. So we set manually arrival_time to  departure_time.
+                     *
+                     * Same reasoning for departure_time.
+                     *
+                     *
+                     * TODO: And this check should be done on Kirin's side...
+                     * */
+                    if (! st.arrival().has_time() && st.departure().has_time()){
+                        arrival_time = departure_time = st.departure().time();
+                    }
+                    if (st.arrival().has_time() && ! st.departure().has_time()){
+                        departure_time = arrival_time =st.arrival().time();
+                    }
+                }
+                auto arrival_seconds = boost::posix_time::from_time_t(arrival_time).time_of_day().total_seconds();
```

Comment:
You should check that arrival_time is after the given day, and then remove  the number of second of the day at midnight to generate the stop_times. Then, if the vj pass midnight, stop times will be greater than 24h, but it's important that the first stop time is between 0 and 24h.